pkg/pillar/hardware/smart.go

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2022. Zededa, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package hardware
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/lf-edge/eve/api/go/info"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+var (
+	maxSmartCtlSize        = 65536
+	idCurrentPendingSector = 197
```

Comment:
Would be good to add a comment indicating where these numbers come from. Is there a document somewhere which defines the smart attribute ids.